### DISCUSSION TITLE
Heterogenous vs. homogeneous reactions

### DISCUSSION THREAD

**Discussion by ma-sadeghi** [PMEAL/OpenPNM#1986] [Ideas]
Currently, when we add a source term (e.g. reaction), we assume that the reaction takes place at pore concentration, which is certainly true for homogeneous reactions.

For heterogeneous reactions, on the other hand, this assumption might be true depending on the relative strength of reaction compared to mass transport.

In the extreme case where the reaction rate is much higher than the mass transport rate, there would be a non-negligible concentration gradient from the center of the pore toward its walls, which inevitably would introduce some error in the calculations. 

I haven't done the math yet, but it seems that assuming idealized shapes for pores, we can come up with a closed-form correction formula.

Not that it's urgent, but I'm just posting for future reference in case we want to go down this road.

**Comment by jgostick** [comment]
This is an excellent point.  The problem I foresee is that mass transfer is a function of flow, and we don't know the flow rate in a pore, so calculating a coefficient will be a challenge.

**Comment by ma-sadeghi** [comment]
You're right. For diffusion-only physics, however, it should be doable I think.
